File: 0896-monotonic-array/0896-monotonic-array.rb
# @param {Integer[]} nums
# @return {Boolean}
def is_monotonic(nums)
    return true if nums.length == 1
    trend = nil
    (1...nums.length).each do |i|
        if nums[i] != nums[i-1]
            trend = nums[i] > nums[i-1] ? 'inc' : 'dec'
            break
        end
    end
    (1...nums.length).each do |i|
        if trend == 'inc'
            return false if nums[i] < nums[i-1]
        elsif trend == 'dec'
            return false if nums[i] > nums[i-1]
        end
    end
    return true
end

File: 0896-monotonic-array/0896-monotonic-array.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var isMonotonic = function(nums) {
    if (nums.length === 1) return true;
    let trend;
    
    for (let i = 1; i < nums.length; i++) {
        if (nums[i] !== nums[i - 1]) {
            trend = nums[i] > nums[i - 1] ? 'inc' : 'dec';
            break;
        }
    }
    
    for (let i = 1; i < nums.length; i++) {
        if (trend === 'inc') {
            if (nums[i] < nums[i - 1]) return false;
        } else if (trend === 'dec') {
            if (nums[i] > nums[i - 1]) return false;
        }
    }
    
    return true;
};